Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/news_classifier.py

```diff
@@ -395,7 +418,18 @@ def get_predictions(self, model, data_loader):
     mlflow.log_param("samples", model.NUM_SAMPLES_COUNT)
     mlflow.log_metric("test_acc", float(test_acc))
     mlflow.log_metric("test_loss", float(test_loss))
-    mlflow.pytorch.log_model(model, "bert-model", registered_model_name="BertModel", extra_files=["class_mapping.json", "bert_base_uncased_vocab.txt"])
+    mlflow.pytorch.log_model(model, "bert-model", registered_model_name=args.model_name, extra_files=["class_mapping.json", "bert_base_uncased_vocab.txt"])
+
+    if args.json_dump is not None:
+        from mlflow.entities.model_registry import ModelVersion
+
+        client = MlflowClient()
+        filter_string = f"run_id='{run_id}'"
+        registered_model: ModelVersion = client.search_model_versions(filter_string=filter_string)[0]
+        model_details = {k.strip("_"): v for k, v in registered_model.__dict__.items()}
```

Comment:
In order to avoid this save and search, wouldn't be better something like this?

```suggestion
    if args.json_dump is not None:
        from mlflow.entities.model_registry import ModelVersion

        client = MlflowClient()
        model_version = mlflow.register_model(f"runs:/{run_id}/artifacts/bert-model", args.model_name)
        model_details = {k.strip("_"): v for k, v in model_version.__dict__.items()}
```

Check here: https://mlflow.org/docs/latest/model-registry.html#adding-an-mlflow-model-to-the-model-registry

Reply:
Perfect, this is much better, we can be much more explicit about registering model this way, and avoids the race conditions/searching